packages/smooth_app/lib/data_models/up_to_date_helper.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:flutter/material.dart';
+
+/// Just an `int` unique identifier, with a distinct class name for clarity.
+@immutable
+abstract class _UpToDateId implements Comparable<_UpToDateId> {
+  const _UpToDateId(this.id);
+
+  final int id;
+
+  @override
+  int get hashCode => id.hashCode;
+
+  @override
+  bool operator ==(Object other) {
+    if (other is! _UpToDateId) {
+      return false;
+    }
+    return id == other.id;
+  }
+
+  @override
+  String toString() => id.toString();
+
+  @override
+  int compareTo(final _UpToDateId other) => id.compareTo(other.id);
+}
```

Comment:
> Wouldn't a typedef be sufficient

@M123-dev Unfortunately no, because my point here is to make obvious that we're not taking about the same class (though they're both `id`s). And the `typedef` does not discriminate:
```dart
typedef Toto = int;
typedef Titi = int;
final Toto toto = 12;
final Titi titi = 12;

void myTest() {
  if (toto == titi) { // I expected to be rejected - not the same class!
    print('same');
  }
}
```